app/javascript/components/set-ownership-form/index.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Grid } from 'patternfly-react';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import { http } from '../../http_api';
+import createSchema from './ownership-form.schema';
+
+class SetOwnershipForm extends Component {
+  state = {
+    initialValues: {},
+    userOptions: [],
+    groupOptions: [],
+  };
+
+  componentDidMount() {
+    const { ownershipIds } = this.props;
+    this.loadInitialData(ownershipIds);
+  }
+
+  addOptions = (ownerShipIds, type, data) => {
+    const ret = [];
+    if (ownerShipIds.length > 1) {
+      ret.push([__('Don\'t change'), 'dont-change']);
+    }
+    ret.push([type === 'user' ? __('No Owner') : __('No User Group'), '']);
+
+    return [...ret, ...data];
+  };
+
+  loadInitialData = (ownershipIds) => {
+    Promise.all([
+      http.get('/api/users?expand=resources'),
+      http.get('/api/groups?expand=resources'),
```

Comment:
```suggestion
      API.get('/api/groups?expand=resources'),
      Promise.new((resolve, reject) => {
        if (ownershipIds.length == 1) {
          // not sure about the structure here, but you need to end the URL with the ID
          // you will definitely have to skip errors as you don't want ugly error modals on 404
          API.get(`/api/tenant_groups/${ownershipIds[0]}`, { skipErrors: true })
            .then(response => resolve(response.id))
            .catch(() => resolve(''));
        } else {
          resolve('');
        }
      });
```

And then if you add an extra argument named `group_id`, you can propagate it in the `groupOptions`.